db/migrate/20211110220355_set_new_collection_associations.rb

```diff
@@ -0,0 +1,33 @@
+# since we have changed the basic association types, we can no longer use the dynamic methods of :branding_group_id or
+# :user_id to reference back to the parent models.  Since the data exists in the documents, we can call :attributes to
+# load existing data and create maps with which to set new associations
+class SetNewCollectionAssociations < Mongoid::Migration
+  def self.up
+    # create maps showing existing associations
+    study_list = ActiveRecordUtils.pluck_to_hash(Study, [:id, :branding_group_id])
+    study_map = study_list.map do |entry|
+      if entry[:branding_group_id].present? && entry[:branding_group_id].is_a?(BSON::ObjectId)
+        { entry[:id].to_s => entry[:branding_group_id] }
+      end
+    end.compact.reduce({}, :merge)
+    collection_map = BrandingGroup.all.map do |col|
+      {
+        col.id.to_s => {
+          users: [User.find(col.attributes[:user_id])],
```

Comment:
I believe `filter` would make the array literal unnecessary, and perhaps be more idiomatic.

```suggestion
          users: User.filter(col.attributes[:user_id]),
```

Reply:
`filter` is not a class method for `User`, so that won't work.  We could do `User.where(id: id).to_a` though, which I kinda like.

Reply:
cfb3f20